File: USACO/speeding.py
import sys
sys.stdin = open("speeding.in", "r")
sys.stdout = open("speeding.out", "w")

road_segments = []
bessie_segments = []
n, m = map(int, input().split())
for _ in range(n):
    miles, limit = map(int, input().split())
    road_segments.extend([limit]*miles)
for _ in range(m):
    miles, limit = map(int, input().split())
    bessie_segments.extend([limit]*miles)

max_over_limit = 0
for limit in range(100):
    if road_segments[limit] < bessie_segments[limit]:
        max_over_limit = max(bessie_segments[limit] - road_segments[limit], max_over_limit)
print(max_over_limit)

File: ____archivedusaco/speeding.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    freopen("speeding.in", "r", stdin);
    freopen("speeding.out", "w", stdout);
    int road_segments[101];
    int bessie_segments[101];
    int n, m; cin >> n >> m;
    for (int i = 0; i < n; i++) {
        int miles, limit;
        cin >> miles >> limit;
        for (int i = 0; i < miles; i++) {
            road_segments[i] = limit;
        }
    }
    
    for (int i = 0; i < m; i++) {
        int miles, limit;
        cin >> miles >> limit;
        for (int i = 0; i < miles; i++) {
            bessie_segments[i] = limit;
        }
    }
    int bad = 0;
    for (int i = 0; i < 100; i++) {
        if (road_segments[i] > bessie_segments[i]) {
            bad = max(bad, road_segments[i]-bessie_segments[i]);
        }
    }
    cout << bad;
}